Sources_v3/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -99,12 +99,13 @@ class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
             let messageDTO = try session.messageEditableByCurrentUser(messageId)
 
             switch messageDTO.localMessageState {
-            case nil:
+            case nil, .pendingSync, .syncingFailed, .deletingFailed:
                 messageDTO.text = text
                 messageDTO.localMessageState = .pendingSync
-            case .pendingSync, .pendingSend:
+            case .pendingSend, .sendingFailed:
                 messageDTO.text = text
-            default:
+                messageDTO.localMessageState = .pendingSend
+            case .sending, .syncing, .deleting:
```

Comment:
I have no idea what's happening here and the commit name "Update `editMessage` logic in `MessageUpdater`" doesn't help either

Reply:
The following change allows to `edit` messages in all the states except `in progress` ones. Without this changes we cannot show `edit` action in the UI for messages failed to be sent/deleted/edited.

Reply:
Linked the description to commit message